src/app/rpmostree-clientlib.cxx

```diff
@@ -831,15 +833,25 @@ rpmostree_transaction_client_run (RpmOstreeCommandInvocation *invocation,
             invocation->exit_code = RPM_OSTREE_EXIT_UNCHANGED;
           return TRUE;
         }
-      else
+      else if (!opt_apply_live)
         {
           /* do diff without dbus: https://github.com/projectatomic/rpm-ostree/pull/116 */
           const char *sysroot_path = rpmostree_sysroot_get_path (sysroot_proxy);
           rpmostreecxx::print_treepkg_diff_from_sysroot_path (rust::Str(sysroot_path),
                 RPMOSTREE_DIFF_PRINT_FORMAT_FULL_MULTILINE, 0, cancellable);
+          g_print ("Changes queued for next boot. Run \"systemctl reboot\" to start a reboot\n");
         }
-
-      g_print ("Run \"systemctl reboot\" to start a reboot\n");
+      else if (opt_apply_live)
+        {
+          const char *sysroot_path = rpmostree_sysroot_get_path (sysroot_proxy);
+          g_autoptr(GFile) sysroot_file = g_file_new_for_path (sysroot_path);
+          g_autoptr(OstreeSysroot) sysroot = ostree_sysroot_new (sysroot_file);
+          if (!ostree_sysroot_load (sysroot, cancellable, error))
+            return FALSE;
+          rpmostreecxx::applylive_finish (*sysroot);
+        }
+      else
+        g_assert_not_reached ();
```

Comment:
Super minor/optional: that `g_assert_not_reached` feels a bit unnecessary given the obviousness of `else if (!opt_apply_live)` and `else if (opt_apply_live)`. I think the diff would read more cleanly if it was just a single insert for the `else if (opt_apply_live)` block before the `else` that deals with the regular treepkg_diff case.

Reply:
I think of it a bit like a Rust `match` on an `enum Action { None, Reboot, ApplyLive }` being represented as two bools because enums are annoying in C - and here we're asserting on the invalid case.